packages/appkit/src/client.ts

```diff
@@ -80,19 +132,38 @@ export class AppKit {
       sdkVersion: SdkVersion
     }
   ) {
-    // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
-    this.adapter = options.adapters?.[0] as ChainAdapter
-    this.caipNetworks = this.extendCaipNetworks(options)
-    this.defaultCaipNetwork = this.extendDefaultCaipNetwork(options)
-    this.initControllers(options)
-    this.initOrContinue()
-    this.version = options.sdkVersion
+    this.options = options
+    this.initialize(options)
   }
 
   public static getInstance() {
     return this.instance
   }
 
+  private async initialize(
+    options: AppKitOptions & {
+      adapters?: ChainAdapter[]
+    } & {
+      sdkVersion: SdkVersion
+    }
+  ) {
+    this.caipNetworks = this.extendCaipNetworks(options)
+
+    this.createAuthProvider()
+    await this.createUniversalProvider()
+    this.createClients()
+    this.defaultCaipNetwork = this.extendDefaultCaipNetwork(options)
+    this.initControllers(options)
+    this.chainAdapters = await this.createAdapters(
+      options.adapters as unknown as AdapterBlueprint[]
+    )
+    await this.initChainAdapters()
+    this.syncRequestedNetworks()
+    await this.initOrContinue()
+    await this.syncExistingConnection()
+    this.version = options.sdkVersion
```

Comment:
do these need to be serial? can we Promise.all any of these to save some time?

Reply:
nit: I'd separate in code the attribute setters first and then the initializer calls whenever possible (just so it's a bit clearer)